### PR TITLE
OWSaveClassifier: Pickle stuff with dill

### PR DESCRIPTION
**Pull request by kernc**
... which supports (nearly) arbitrary Python objects.
Fixes https://github.com/biolab/orange3/issues/1758

##### Includes
- [X] Code changes
- [x] Tests

### CODE REVIEW

**Comment by astaric** [comment]
Can you include some new test that show that dill can pickle lambdas? focus on lambdas that use variables from different scopes.

**Comment by astaric** [comment]
And tests that show that dill can "unpickle" saved classifiers created with pickle.

**Comment by kernc** [comment]
Done.

**Comment by ajdapretnar** [comment]
Getting a crash if using a classifier, built on a different domain. Should get an error. Just writing it down here, not meant as a part of this PR. I'm opening it as a new issue.

p.s. I also got some recurring error (log just wouldn't go away and had to force-close the program), but I cannot reproduce it currently.

File: Orange/widgets/data/tests/test_owfeatureconstructor.py [hunk 1]
```diff
@@ -71,6 +73,27 @@ def test_construct_variables_string(self):
...
+
+        lambda_func = lambda x, LOCAL_CONST=7: \
+            x * LOCAL_CONST * NONLOCAL_CONST * self.CLASS_CONST * GLOBAL_CONST
+
+        def nested_func(x, LOCAL_CONST=7):
+            return x * LOCAL_CONST * NONLOCAL_CONST * self.CLASS_CONST * GLOBAL_CONST
+
+        self.assertEqual(lambda_func(11),
+                         pickle.loads(pickle.dumps(lambda_func))(11))
+        self.assertEqual(nested_func(11),
```
Comment (kernc) [inline Orange/widgets/data/tests/test_owfeatureconstructor.py]: Interestingly, it remembers the exact values of locals and nonlocals at pickling, so the computed result after unpickling is different. Changing the global after pickling errors with:
```
Traceback (most recent call last):
  File "/home/jk/PycharmProjects/orange3/Orange/widgets/data/tests/test_owfeatureconstructor.py", line 94, in test_lambdas_pickle
    pickled_lambda = pickle.dumps(lambda_func)
  File "/home/jk/Downloads/dill/dill/dill.py", line 256, in dumps
    dump(obj, file, protocol, byref, fmode, recurse)#, strictio)
  File "/home/jk/Downloads/dill/dill/dill.py", line 249, in dump
    pik.dump(obj)
  File "/usr/lib/python3.4/pickle.py", line 412, in dump
    self.save(obj)
  File "/usr/lib/python3.4/pickle.py", line 479, in save
    f(self, obj) # Call unbound method with explicit self
  File "/home/jk/Downloads/dill/dill/dill.py", line 1268, in save_function
    globs = globalvars(obj, recurse=True, builtin=True)
  File "/home/jk/Downloads/dill/dill/detect.py", line 200, in globalvars
    _vars = globalvars(obj.cell_contents, recurse, builtin) or {}
ValueError: Cell is empty
```
It works if a global is defined `global`, but this statement is impossible in lambdas. The difference `global` statement makes is in `LOAD_DEREF` vs `LOAD_GLOBAL` opcodes. I don't know if it is or will become an issue in a practical case.

File: Orange/widgets/classify/tests/test_owloadclassifier.py [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 # pylint: disable=missing-docstring
 import os
-import pickle
 from tempfile import mkstemp
 
 from Orange.classification.majority import ConstantModel
 from Orange.widgets.classify.owloadclassifier import OWLoadClassifier
 from Orange.widgets.tests.base import WidgetTest
 
+import pickle
```
Comment (lanzagar) [inline Orange/widgets/classify/tests/test_owloadclassifier.py]: A comment here would be nice, to explain why this needs to be after Orange imports and not grouped with python imports at the top.

**Pull request merged**